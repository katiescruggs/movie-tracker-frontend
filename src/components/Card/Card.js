import React from 'react';
import PropTypes from 'prop-types';
import * as actions from '../../actions';
import { connect } from 'react-redux';
import VideoCamera from 'react-icons/lib/fa/video-camera';
import './Card.css';

const Card = ({userId, addFav, removeFav, movie, favorites, path}) => {
  const isFav = favorites.find(fav => (fav.title === movie.title));
  
  const favoriteClick = () => {
    const movieId = path === '/favorites' ? movie.movie_id : movie.id;
    isFav ? removeFav(userId, movieId) : addFav(userId, movie);
  };

  const {title, poster_path, release_date, vote_average, overview} = movie;
  const buttonClass = isFav ? 'btn-card-favorite fav' : 'btn-card-favorite';

  return (
    <div className='card-wrapper'>
      <div className="card">
        <div className='flipper'>
          <div className='front'>
            <img src={`https://image.tmdb.org/t/p/w500${poster_path}`} />
          </div>
          <div className='back'>
            <header>
              <h3>{title}</h3>
            </header>
            <div className='card-body'>
              <p>
                <span className='release-date'>
                  Release Date: 
                </span>{release_date}
              </p>
              <p>
                <span className='vote'>
                  Vote Average: 
                </span>
                {vote_average}
              </p>
              <p>
                <span className='overview'>
                  Overview: 
                </span>
                {overview}
              </p>
            </div>
          </div>
        </div>
      </div>
      <button 
        onClick={favoriteClick} 
        className={buttonClass}
      >
        FAVORITE 
        <VideoCamera className='video-icon'/>
      </button>
    </div>
  );
};

Card.propTypes = {
  movie: PropTypes.object,
  userId: PropTypes.number,
  addFav: PropTypes.func,
  removeFav: PropTypes.func,
  favorites: PropTypes.array,
  path: PropTypes.string
};

const mapStateToProps = state => ({
  userId: state.user.info.id,
  favorites: state.favorites
});

const mapDispatchToProps = dispatch => {
  return {
    addFav: (userId, movie) => {
      dispatch(actions.addFavorite(userId, movie));
    },
    removeFav: (userId, favId) => {
      dispatch(actions.removeFavorite(userId, favId));
    }
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Card);