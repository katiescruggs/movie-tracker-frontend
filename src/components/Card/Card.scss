.card  {
  img {
    width: 300px;
  }
  // background: white;
  margin: 50px;
  perspective: 1000px;
}


.card:hover .flipper,
.card.hover .flipper {
  transform: rotateY(180deg);
}

.card,
.front,
.back {
  height: 450px;
  width: 300px;
}

.flipper {
  transition: 1s;
  transform-style: preserve-3d;
  position: relative;
  transform-origin: 150px 100%;
}

.front, 
.back {
  backface-visibility: hidden;
  position: absolute;
  top: 0;
  left: 0;
}

.front {
  z-index: 2;
  transform: rotateY(0deg);
}

.back {
  & p {
      margin: 5px 0;
      padding: 0px 25px;
  }
& header {
  background: #43465c;
  padding: 20px;
  font-size: 16pt;
  color: #fff;
}
  transform: rotateY(180deg);
  background: #DEDEDE;

}

.back .card-body {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  text-align: left;
}

.btn-card-favorite {
  font-size: 16pt;
  padding: 8px;
  border-radius: 15px;
}

.fav {
  border: 10px solid magenta;
}

.star-icon {
}