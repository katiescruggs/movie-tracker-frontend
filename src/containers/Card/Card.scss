.card  {
  img {
    height: 450px;
    width: 300px;
  }
  margin: 70px 30px 0;
  perspective: 1000px;
}

.card:hover .flipper,
.card.hover .flipper {
  transform: rotateY(180deg);
}

.card,
.front,
.back {
  height: 450px;
  width: 300px;
}

.flipper {
  transition: 1s;
  transform-style: preserve-3d;
  position: relative;
  transform-origin: 150px 100%;
}

.front, 
.back {
  backface-visibility: hidden;
  position: absolute;
  top: 0;
  left: 0;
  box-shadow: 0px 0px 5px 2px rgba(255,207,90,0.6);
}

.front {
  z-index: 2;
  transform: rotateY(0deg);
}

.back {
  & p {
      margin: 10px 0;
      padding: 0px 25px;
  }
& header {
  background-image: url('../../Assets/stage-lights.jpg');
  background-position: center;
  padding: 20px;
  font-size: 16pt;
  color: #fff;
  height: 100px;
  display: flex;
  justify-content: center;
  align-items: center;
}
  transform: rotateY(180deg);
  background: #DEDEDE;

}
.vote, .release-date, .overview {
  font-weight: 700;
  margin-right: 4px;
  color: #1c285b;
}

.back .card-body {
  text-align: left;
  overflow: scroll;
  height: 350px;
}

.btn-card-favorite {
  border: none;
  font-size: 16pt;
  padding: 10px;
  background: rgba(0, 0, 0, 0.5);
  color: white;
  outline: none;
}

.btn-card-favorite:hover {
  color: #FFE9A8;
  cursor: pointer;
}

.fav {
  color: #ffcf40;
}

.video-icon {
  margin: 0 0 5px 10px;
}

.card-wrapper {
  text-align: center;
}