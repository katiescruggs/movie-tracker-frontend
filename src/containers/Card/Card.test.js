import React from 'react';
import { shallow } from 'enzyme';
import { Card, mapStateToProps, mapDispatchToProps } from './Card';

describe('Card tests', () => {
  describe('Card', () => {
    let card;
    const mockAddFav = jest.fn();
    const mockRemoveFav = jest.fn();
    const mockMovie = {
      title: 'title',
      poster_path: '.asdf',
      release_date: 'date',
      vote_average: '2',
      overview: 'overview'
    };
    let mockFavorites = ['array', 'of', 'favorites'];

    beforeEach( () => {
      card = shallow(
        <Card 
          userId={1}
          addFav={mockAddFav}
          removeFav={mockRemoveFav}
          movie={mockMovie}
          favorites={mockFavorites}
          path={'/favorites'}
        />);
    });

    it('should exist', () => {
      expect(card).toBeDefined();
    });

    it('should match the snapshot', () => {
      expect(card).toMatchSnapshot();
    });

    it('should call addFav on click', () => {
      card.find('.btn-card-favorite').first().simulate('click');
      expect(mockAddFav).toHaveBeenCalled();
    });

    it('should call removeFav on click', () => {
      mockFavorites = [{title: 'title'}];
      card = shallow(
        <Card 
          userId={1}
          addFav={mockAddFav}
          removeFav={mockRemoveFav}
          movie={mockMovie}
          favorites={mockFavorites}
          path={'/favorites'}
        />);

      card.find('.btn-card-favorite').first().simulate('click');
      expect(mockRemoveFav).toHaveBeenCalled();
    });
  });

  describe('mapStateToProps', () => {
    const mockStore = {
      user: {
        info: {
          id: 0
        }
      },
      favorites: ['array', 'of', 'favorites']
    };
    const result = mapStateToProps(mockStore);

    it('should pull userId from the store', () => {
      expect(result.userId).toEqual(mockStore.user.info.id);
    });

    it('should pull favorites from the store', () => {
      expect(result.favorites).toEqual(mockStore.favorites);
    });
  });

  describe('mapDispatchToProps', () => {
    let mockDispatch;
    let result;

    beforeEach(() => {
      mockDispatch = jest.fn();
      result = mapDispatchToProps(mockDispatch);
    });

    it('should call dispatch when addFav is called', () => {
      result.addFav();
      expect(mockDispatch).toHaveBeenCalled();
    });

    it('should call dispatch when removeFav is called', () => {
      result.removeFav();
      expect(mockDispatch).toHaveBeenCalled();
    });
  });
});