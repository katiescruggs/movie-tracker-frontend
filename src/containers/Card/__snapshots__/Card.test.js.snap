// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card tests Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    addFav={[Function]}
    favorites={
        Array [
            "array",
            "of",
            "favorites",
          ]
    }
    movie={
        Object {
            "overview": "overview",
            "poster_path": ".asdf",
            "release_date": "date",
            "title": "title",
            "vote_average": "2",
          }
    }
    path="/favorites"
    removeFav={[Function]}
    userId={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="card"
>
          <div
                    className="flipper"
          >
                    <div
                              className="front"
                    >
                              <img
                                        src="https://image.tmdb.org/t/p/w500.asdf"
                              />
                    </div>
                    <div
                              className="back"
                    >
                              <header>
                                        <h3>
                                                  title
                                        </h3>
                              </header>
                              <div
                                        className="card-body"
                              >
                                        <p>
                                                  <span
                                                            className="release-date"
                                                  >
                                                            Release Date:
                                                  </span>
                                                  date
                                        </p>
                                        <p>
                                                  <span
                                                            className="vote"
                                                  >
                                                            Vote Average:
                                                  </span>
                                                  2
                                        </p>
                                        <p>
                                                  <span
                                                            className="vote"
                                                  >
                                                            Overview:
                                                  </span>
                                                  overview
                                        </p>
                              </div>
                    </div>
          </div>
</div>,
        <button
          className="btn-card-favorite"
          onClick={[Function]}
>
          FAVORITE
          <FaVideoCamera
                    className="video-icon"
          />
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="flipper"
>
            <div
                        className="front"
            >
                        <img
                                    src="https://image.tmdb.org/t/p/w500.asdf"
                        />
            </div>
            <div
                        className="back"
            >
                        <header>
                                    <h3>
                                                title
                                    </h3>
                        </header>
                        <div
                                    className="card-body"
                        >
                                    <p>
                                                <span
                                                            className="release-date"
                                                >
                                                            Release Date:
                                                </span>
                                                date
                                    </p>
                                    <p>
                                                <span
                                                            className="vote"
                                                >
                                                            Vote Average:
                                                </span>
                                                2
                                    </p>
                                    <p>
                                                <span
                                                            className="vote"
                                                >
                                                            Overview:
                                                </span>
                                                overview
                                    </p>
                        </div>
            </div>
</div>,
          "className": "card",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="front"
>
                <img
                                src="https://image.tmdb.org/t/p/w500.asdf"
                />
</div>,
              <div
                className="back"
>
                <header>
                                <h3>
                                                title
                                </h3>
                </header>
                <div
                                className="card-body"
                >
                                <p>
                                                <span
                                                                className="release-date"
                                                >
                                                                Release Date:
                                                </span>
                                                date
                                </p>
                                <p>
                                                <span
                                                                className="vote"
                                                >
                                                                Vote Average:
                                                </span>
                                                2
                                </p>
                                <p>
                                                <span
                                                                className="vote"
                                                >
                                                                Overview:
                                                </span>
                                                overview
                                </p>
                </div>
</div>,
            ],
            "className": "flipper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  src="https://image.tmdb.org/t/p/w500.asdf"
/>,
                "className": "front",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "src": "https://image.tmdb.org/t/p/w500.asdf",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <header>
                    <h3>
                                        title
                    </h3>
</header>,
                  <div
                    className="card-body"
>
                    <p>
                                        <span
                                                            className="release-date"
                                        >
                                                            Release Date:
                                        </span>
                                        date
                    </p>
                    <p>
                                        <span
                                                            className="vote"
                                        >
                                                            Vote Average:
                                        </span>
                                        2
                    </p>
                    <p>
                                        <span
                                                            className="vote"
                                        >
                                                            Overview:
                                        </span>
                                        overview
                    </p>
</div>,
                ],
                "className": "back",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <h3>
                      title
</h3>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "title",
                    },
                    "ref": null,
                    "rendered": "title",
                    "type": "h3",
                  },
                  "type": "header",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <p>
                        <span
                                                className="release-date"
                        >
                                                Release Date:
                        </span>
                        date
</p>,
                      <p>
                        <span
                                                className="vote"
                        >
                                                Vote Average:
                        </span>
                        2
</p>,
                      <p>
                        <span
                                                className="vote"
                        >
                                                Overview:
                        </span>
                        overview
</p>,
                    ],
                    "className": "card-body",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          <span
                            className="release-date"
>
                            Release Date:
</span>,
                          "date",
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "Release Date:",
                            "className": "release-date",
                          },
                          "ref": null,
                          "rendered": "Release Date:",
                          "type": "span",
                        },
                        "date",
                      ],
                      "type": "p",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          <span
                            className="vote"
>
                            Vote Average:
</span>,
                          "2",
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "Vote Average:",
                            "className": "vote",
                          },
                          "ref": null,
                          "rendered": "Vote Average:",
                          "type": "span",
                        },
                        "2",
                      ],
                      "type": "p",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          <span
                            className="vote"
>
                            Overview:
</span>,
                          "overview",
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "Overview:",
                            "className": "vote",
                          },
                          "ref": null,
                          "rendered": "Overview:",
                          "type": "span",
                        },
                        "overview",
                      ],
                      "type": "p",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "FAVORITE",
            <FaVideoCamera
              className="video-icon"
/>,
          ],
          "className": "btn-card-favorite",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "FAVORITE",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "video-icon",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card"
>
            <div
                        className="flipper"
            >
                        <div
                                    className="front"
                        >
                                    <img
                                                src="https://image.tmdb.org/t/p/w500.asdf"
                                    />
                        </div>
                        <div
                                    className="back"
                        >
                                    <header>
                                                <h3>
                                                            title
                                                </h3>
                                    </header>
                                    <div
                                                className="card-body"
                                    >
                                                <p>
                                                            <span
                                                                        className="release-date"
                                                            >
                                                                        Release Date:
                                                            </span>
                                                            date
                                                </p>
                                                <p>
                                                            <span
                                                                        className="vote"
                                                            >
                                                                        Vote Average:
                                                            </span>
                                                            2
                                                </p>
                                                <p>
                                                            <span
                                                                        className="vote"
                                                            >
                                                                        Overview:
                                                            </span>
                                                            overview
                                                </p>
                                    </div>
                        </div>
            </div>
</div>,
          <button
            className="btn-card-favorite"
            onClick={[Function]}
>
            FAVORITE
            <FaVideoCamera
                        className="video-icon"
            />
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="flipper"
>
              <div
                            className="front"
              >
                            <img
                                          src="https://image.tmdb.org/t/p/w500.asdf"
                            />
              </div>
              <div
                            className="back"
              >
                            <header>
                                          <h3>
                                                        title
                                          </h3>
                            </header>
                            <div
                                          className="card-body"
                            >
                                          <p>
                                                        <span
                                                                      className="release-date"
                                                        >
                                                                      Release Date:
                                                        </span>
                                                        date
                                          </p>
                                          <p>
                                                        <span
                                                                      className="vote"
                                                        >
                                                                      Vote Average:
                                                        </span>
                                                        2
                                          </p>
                                          <p>
                                                        <span
                                                                      className="vote"
                                                        >
                                                                      Overview:
                                                        </span>
                                                        overview
                                          </p>
                            </div>
              </div>
</div>,
            "className": "card",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="front"
>
                  <img
                                    src="https://image.tmdb.org/t/p/w500.asdf"
                  />
</div>,
                <div
                  className="back"
>
                  <header>
                                    <h3>
                                                      title
                                    </h3>
                  </header>
                  <div
                                    className="card-body"
                  >
                                    <p>
                                                      <span
                                                                        className="release-date"
                                                      >
                                                                        Release Date:
                                                      </span>
                                                      date
                                    </p>
                                    <p>
                                                      <span
                                                                        className="vote"
                                                      >
                                                                        Vote Average:
                                                      </span>
                                                      2
                                    </p>
                                    <p>
                                                      <span
                                                                        className="vote"
                                                      >
                                                                        Overview:
                                                      </span>
                                                      overview
                                    </p>
                  </div>
</div>,
              ],
              "className": "flipper",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    src="https://image.tmdb.org/t/p/w500.asdf"
/>,
                  "className": "front",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "src": "https://image.tmdb.org/t/p/w500.asdf",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <header>
                      <h3>
                                            title
                      </h3>
</header>,
                    <div
                      className="card-body"
>
                      <p>
                                            <span
                                                                  className="release-date"
                                            >
                                                                  Release Date:
                                            </span>
                                            date
                      </p>
                      <p>
                                            <span
                                                                  className="vote"
                                            >
                                                                  Vote Average:
                                            </span>
                                            2
                      </p>
                      <p>
                                            <span
                                                                  className="vote"
                                            >
                                                                  Overview:
                                            </span>
                                            overview
                      </p>
</div>,
                  ],
                  "className": "back",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <h3>
                        title
</h3>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "title",
                      },
                      "ref": null,
                      "rendered": "title",
                      "type": "h3",
                    },
                    "type": "header",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <p>
                          <span
                                                    className="release-date"
                          >
                                                    Release Date:
                          </span>
                          date
</p>,
                        <p>
                          <span
                                                    className="vote"
                          >
                                                    Vote Average:
                          </span>
                          2
</p>,
                        <p>
                          <span
                                                    className="vote"
                          >
                                                    Overview:
                          </span>
                          overview
</p>,
                      ],
                      "className": "card-body",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            <span
                              className="release-date"
>
                              Release Date:
</span>,
                            "date",
                          ],
                        },
                        "ref": null,
                        "rendered": Array [
                          Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "children": "Release Date:",
                              "className": "release-date",
                            },
                            "ref": null,
                            "rendered": "Release Date:",
                            "type": "span",
                          },
                          "date",
                        ],
                        "type": "p",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            <span
                              className="vote"
>
                              Vote Average:
</span>,
                            "2",
                          ],
                        },
                        "ref": null,
                        "rendered": Array [
                          Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "children": "Vote Average:",
                              "className": "vote",
                            },
                            "ref": null,
                            "rendered": "Vote Average:",
                            "type": "span",
                          },
                          "2",
                        ],
                        "type": "p",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            <span
                              className="vote"
>
                              Overview:
</span>,
                            "overview",
                          ],
                        },
                        "ref": null,
                        "rendered": Array [
                          Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "children": "Overview:",
                              "className": "vote",
                            },
                            "ref": null,
                            "rendered": "Overview:",
                            "type": "span",
                          },
                          "overview",
                        ],
                        "type": "p",
                      },
                    ],
                    "type": "div",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "FAVORITE",
              <FaVideoCamera
                className="video-icon"
/>,
            ],
            "className": "btn-card-favorite",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "FAVORITE",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "video-icon",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
